### PR TITLE
HumanoidBench speed up ~30x

### PR DESCRIPTION
**Pull request by geng-haoran**
We speed up humanoidbench to around 30x.

First we make get_state as batched tensor
Then we merge humanoidbench tasks with batch reward, checker and so on.

It speed up training around 30x

### CODE REVIEW

File: metasim/cfg/tasks/humanoidbench/base_cfg.py [hunk 1]
```diff
@@ -125,30 +101,52 @@ def __init__(self, robot_name="h1"):
...
+            sigmoid="linear",
+        )
+        stand_reward = standing * upright
+        small_control = humanoid_reward_util.tolerance_tensor(
+            actuator_forces_tensor(states, self.robot_name),
+            margin=10,
+            value_at_margin=0,
+            sigmoid="quadratic",
+        ).mean()
+        small_control = (4 + small_control) / 5
```
Comment (Fisher-Wang) [inline metasim/cfg/tasks/humanoidbench/base_cfg.py:123]: feel free to delete these commented lines

File: metasim/cfg/tasks/humanoidbench/base_cfg.py [hunk 2]
```diff
@@ -168,21 +166,36 @@ def __call__(self, states: list[EnvState]) -> torch.FloatTensor:
...
+            move = humanoid_reward_util.tolerance_tensor(
+                com_velocity,
+                bounds=(self._move_speed, float("inf")),
+                margin=self._move_speed,
+                value_at_margin=0,
+                sigmoid="linear",
+            )
+            move = (5 * move + 1) / 6
+            moving_reward = move
+        # for state in states:
```
Comment (Fisher-Wang) [inline metasim/cfg/tasks/humanoidbench/base_cfg.py:184]: feel free to delete

File: metasim/utils/humanoid_robot_util.py [hunk 3]
```diff
@@ -120,3 +157,42 @@ def quaternion_to_matrix(quat: torch.Tensor):
         dtype=quat.dtype,
     )
     return R
+
+
+def quaternion_to_matrix_tensor(quat: torch.Tensor) -> torch.Tensor:
```
Comment (Fisher-Wang) [inline metasim/utils/humanoid_robot_util.py:162]: please use the function `matrix_from_quat` in `metasim/utils/math.py` instead

File: metasim/utils/humanoid_robot_util.py [hunk 4]
```diff
@@ -9,6 +9,14 @@ def torso_upright(envstate, robot_name: str):
     return xmat[2, 2]
 
 
+def torso_upright_tensor(envstate, robot_name: str):
+    """Returns projection from z-axes of torso to the z-axes of world."""
+    robot_body_name = envstate.robots[robot_name].body_names
```
Comment (Fisher-Wang) [inline metasim/utils/humanoid_robot_util.py:14]: Please add a "TODO", "XXX", or "FIXME" here since it is not a standard behavior

File: metasim/utils/humanoid_robot_util.py [hunk 5]
```diff
@@ -25,6 +33,16 @@ def neck_height(envstate, robot_name: str):
     ) / 2
 
 
+def neck_height_tensor(envstate, robot_name: str):
+    """Returns the height of the neck."""
+    robot_body_name = envstate.robots[robot_name].body_names
```
Comment (Fisher-Wang) [inline metasim/utils/humanoid_robot_util.py:38]: Please add a "TODO", "XXX", or "FIXME" here since it is not a standard behavior

File: roboverse_learn/humanoidbench_rl/wrapper_sb3.py [hunk 6]
```diff
@@ -206,17 +205,18 @@ def step_wait(self):
...
 
     def get_humanoid_reward(self, states):
         # NOTE: For IsaacLab, metasim_reward is None, so calculate reward here
-        states = state_tensor_to_nested(self.env.handler, states)
-        final_reward = torch.zeros(self.num_envs)
+        final_reward = torch.zeros(self.num_envs, device=self.sim_device)
         for reward_func, reward_weight in zip(
             self.env.handler.task.reward_functions, self.env.handler.task.reward_weights
         ):
+            states.robots[self.env.handler.robot.name].body_names = self.env.handler.get_body_names(
```
Comment (Fisher-Wang) [inline roboverse_learn/humanoidbench_rl/wrapper_sb3.py:217]: Please add a "TODO", "XXX", or "FIXME" here since it is not a standard behavior

**Comment by Fisher-Wang** [comment]
@EmptyBlueBox Could you also review this PR since it is related to your part?

**Review by EmptyBlueBox — COMMENTED** [review]
Humanoidbench part looks good for me!
